### PR TITLE
HBASE-27567 Introduce ChaosMonkey Action to print HDFS Cluster status

### PR DESCRIPTION
**Pull request by ndimiduk**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 20s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 26s |  master passed  |
| +1 :green_heart: |  compile  |   0m 17s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 17s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 45s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 39s |  hbase-it in the patch passed.  |
|  |   |  16m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 692d20d005eb 5.4.0-135-generic #152-Ubuntu SMP Wed Nov 23 20:19:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c1b9d8ba06 |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/testReport/ |
| Max. process+thread count | 556 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 55s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 35s |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 23s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 23s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 43s |  hbase-it in the patch passed.  |
|  |   |  20m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8bd25e3829d9 5.4.0-1088-aws #96~18.04.1-Ubuntu SMP Mon Oct 17 02:57:48 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c1b9d8ba06 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/testReport/ |
| Max. process+thread count | 589 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 48s |  master passed  |
| +1 :green_heart: |  compile  |   0m 28s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 25s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   8m 46s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 40s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  24m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 18527f164981 5.4.0-131-generic #147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c1b9d8ba06 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 15s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 10s |  master passed  |
| +1 :green_heart: |  compile  |   0m 21s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 40s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 43s |  hbase-it in the patch passed.  |
|  |   |  20m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 71759c9cdd6c 5.4.0-1088-aws #96~18.04.1-Ubuntu SMP Mon Oct 17 02:57:48 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 2a7c69d30e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/2/testReport/ |
| Max. process+thread count | 597 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 29s |  master passed  |
| +1 :green_heart: |  compile  |   0m 26s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 25s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   8m 45s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  24m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 89673369bd76 5.4.0-131-generic #147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 2a7c69d30e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 80 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Reidddddd — APPROVED** [review]

File: hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActiveNameNodeAction.java [hunk 1]
```diff
@@ -57,39 +57,47 @@ protected Logger getLogger() {
...
+            activeNamenode = proto.getHostname();
+            activeNamenodePort = proto.getPort();
+          }
+        }
       }
     }
+
     if (activeNamenode == null) {
       throw new Exception("No active Name node found in zookeeper under " + hadoopHAZkNode);
+    } else {
```
Comment (Apache9) [inline hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActiveNameNodeAction.java:96]: Why an else here? Seems not necessary,,,

File: hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/DumpHdfsClusterStatusAction.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      sb.append('\n');
+      dns = dfs.getClient().datanodeReport(HdfsConstants.DatanodeReportType.DEAD);
+      sb.append("Number of dead DataNodes: ").append(dns.length);
+      for (DatanodeInfo dni : dns) {
+        sb.append(PREFIX).append(dni.getName()).append("/").append(dni.getNetworkLocation());
+      }
+    }
+    // TODO: add more on NN, JNs, and ZK.
+    // TODO: Print how long process has been up.
+    LOG.info(sb.toString());
```
Comment (Apache9) [inline hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/DumpHdfsClusterStatusAction.java:76]: getLogger().info?

File: hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActiveNameNodeAction.java [hunk 3]
```diff
@@ -57,39 +57,47 @@ protected Logger getLogger() {
...
+            ActiveNodeInfo proto = ActiveNodeInfo.parseFrom(data);
+            activeNamenode = proto.getHostname();
+            activeNamenodePort = proto.getPort();
+          }
+        }
       }
     }
+
     if (activeNamenode == null) {
       throw new Exception("No active Name node found in zookeeper under " + hadoopHAZkNode);
```
Comment (ndimiduk) [inline hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RestartActiveNameNodeAction.java:95]: this could just return instead of throwing.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 42s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 21s |  master passed  |
| +1 :green_heart: |  compile  |   0m 14s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m  1s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 37s |  hbase-it in the patch passed.  |
|  |   |  15m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0fb6e65bd6bd 5.4.0-1088-aws #96~18.04.1-Ubuntu SMP Mon Oct 17 02:57:48 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 2a7c69d30e |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/testReport/ |
| Max. process+thread count | 559 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 39s |  master passed  |
| +1 :green_heart: |  compile  |   0m 23s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 47s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 19s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 13s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  hbase-it in the patch passed.  |
|  |   |  19m 52s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 74a555a7426f 5.4.0-131-generic #147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 2a7c69d30e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/testReport/ |
| Max. process+thread count | 589 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 41s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 58s |  master passed  |
| +1 :green_heart: |  compile  |   0m 26s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 25s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 36s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 24s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 24s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  10m  7s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 43s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 29s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  26m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4959 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 13339363f0e7 5.4.0-1092-aws #100~18.04.2-Ubuntu SMP Tue Nov 29 08:39:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 2a7c69d30e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 80 (vs. ulimit of 30000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4959/3/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ndimiduk** [comment]
Do you have any further concerns @Apache9 ?

**Review by Apache9 — APPROVED** [review]
I was wondering what if we run ITs against S3AFileSystem, but since in ServerAndDependenciesKillingMonkeyFactory, we have already add RestartRandomDataNodeAction which is also HDFS specific, the PR here does not introduce new problems.

**Comment by ndimiduk** [comment]
We could run Chaos tests against a cluster configured with S3A storage, but in that case we would not run the HDFS chaos components.

I'd like to have a declarative configuration-driven chaos definition, instead of using these hard-coded factories, but I don't think that's been implemented.

**Pull request merged**